src/main.rs

```diff
@@ -512,3 +538,246 @@ fn setup() -> Opts {
+            output: output_path,
+        };
+        return_opts
+    }
+
+    /// Tests a single fq file for a seq starting with either A or G
+    ///
+    /// Workflow
+    /// Define a temp dir
+    /// Define the sequences to search in
+    /// Define the pattern to seach for
+    /// Call call_opts() and pass resulting StructOpts instance to fqgrep_from_opts()
+    /// Assert that fqgrep() returns two matches
+    ///  
+    #[test]
+    fn test_single_fq() {
+        let dir = TempDir::new().unwrap();
+        let seqs = vec![vec![
+            "AAGTCTGAATCCATGGAAAGCTATTG",
+            "GGGTCTGAATCCATGGAAAGCTATTG",
+        ]];
+        let test_pattern = vec![String::from("^A"), String::from("^G")];
+        let mut opts_testcase = call_opts(&dir, &seqs, &test_pattern, false, None);
+        // TODO: Automatically set count as t/f depending on opts.output
+        opts_testcase.count = true;
+        let result = fqgrep_from_opts(&mut opts_testcase);
+        assert_eq!(result.unwrap(), 2)
+    }
+
+    /// Tests two files (paired reads) for seqs starting with either A or G
+    ///
+    /// Workflow
+    /// Define a temp dir
+    /// Define the sequences to search in
+    /// Define the pattern to seach for
+    /// Call call_opts() and pass resulting StructOpts instance to fqgrep_from_opts()
+    /// Set opts.testcase.paired to true
+    /// Assert that fqgrep() returns two matches
+    ///
+    #[test]
+    fn test_multiple_fq() {
+        let dir = TempDir::new().unwrap();
+        let seqs = vec![
+            vec!["AAGTCTGAATCCATGGAAAGCTATTG", "GGGTCTGAATCCATGGAAAGCTATTG"],
+            vec!["AAGTCTGAATCCATGGAAAGCTATTG", "GGGTCTGAATCCATGGAAAGCTATTG"],
+        ];
+        let test_pattern = vec![String::from("^A"), String::from("^G")];
+        let mut opts_testcase = call_opts(&dir, &seqs, &test_pattern, true, None);
+        opts_testcase.count = true;
+        opts_testcase.paired = true;
+        let result = fqgrep_from_opts(&mut opts_testcase);
+        assert_eq!(result.unwrap(), 2)
+    }
+
+    /// Tests a two fastqs for a seq that starts with A
+    ///
+    /// Workflow
+    /// Set io as io from fgoxide::io::Io
+    /// Define a temp dir and add an output path to the dir
+    /// Clone the output path so it can be read from
+    ///
+    /// Define the sequences to search in
+    /// Define the pattern to seach for
+    /// Call call_opts() and pass resulting StructOpts instance to fqgrep_from_opts()
+    /// Assert that fqgrep() returns two matches
+    ///
+    #[test]
+    fn test_get_output() {
+        let io = Io::default();
+        let dir: TempDir = TempDir::new().unwrap();
+        let out_path = dir.path().join(String::from("output.fq"));
+        // Clone path to read from below
+        // TODO: refactor
+        let x = &out_path.clone();
+        let seqs = vec![
+            vec!["AAGTCTGAATCCATGGAAAGCTATTG"],
+            vec!["GGGTCTGAATCCATGGAAAGCTATTG"],
+        ];
```

Comment:
Add seq that wont match
```suggestion
        let seqs = vec![
            vec!["AAGTCTGAATCCATGGAAAGCTATTG"],
            vec!["GGGTCTGAATCCATGGAAAGCTATTG"],
            vec!["CGGGGGGGGGGGTTTTTTTTTTTTTT"],
        ];
```